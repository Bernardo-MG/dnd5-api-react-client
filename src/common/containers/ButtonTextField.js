import React, { Component } from 'react';

import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';

import TextField from '@material-ui/core/TextField';

class ButtonTextField extends Component {

   state = { query: '' };

   /**
    * Handles the value change event.
    *
    * @param event value change event
    */
   _onHandleChange = (event) => {
      this.setState({ query: event.target.value });
   };

   /**
    * Handles the button click event.
    *
    * @param event button click event
    */
   _onHandleClick = () => {
      if (this.state.query) {
         this.props.onClick(this.state.query);
      }
   };

   render() {
      return (
         <React.Fragment>
            <TextField
               id={this.props.id}
               label={this.props.label}
               value={this.state.query}
               onChange={::this._onHandleChange}
            />
            <Button variant='contained' onClick={::this._onHandleClick}>
               {this.props.buttonLabel}
            </Button>
         </React.Fragment>
      );
   }

}

ButtonTextField.propTypes = {
   id: PropTypes.string.isRequired,
   label: PropTypes.string.isRequired,
   buttonLabel: PropTypes.string.isRequired,
   onClick: PropTypes.func.isRequired
};

export default ButtonTextField;
