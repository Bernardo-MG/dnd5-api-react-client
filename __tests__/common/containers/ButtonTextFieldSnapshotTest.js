import React from 'react'
import renderer from 'react-test-renderer';

import ButtonTextField from 'common/containers/ButtonTextField';

describe('<ButtonTextField />', () => {
   it('renders correctly', () => {
      const tree = renderer
         .create(<ButtonTextField
               id='testButtonTextField'
               label='label'
               buttonLabel='button label'
               onClick={ () => 'action' } />)
         .toJSON();
      expect(tree).toMatchSnapshot();
   })
});
