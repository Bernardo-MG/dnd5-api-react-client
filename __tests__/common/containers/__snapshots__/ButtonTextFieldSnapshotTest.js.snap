// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonTextField /> renders correctly 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      htmlFor="testButtonTextField"
    >
      label
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="testButtonTextField"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      button label
    </span>
  </button>,
]
`;
